client/OPUNetGameSelectWnd.cpp

```diff
@@ -303,33 +303,40 @@ void OPUNetGameSelectWnd::OnDestroy()
 		KillTimer(this->hWnd, timer);
 	}
 
-	// Save the PlayerName
-	char keyValue[MaxPlayerNameLength];
-	GetDlgItemText(this->hWnd, IDC_PlayerName, keyValue, sizeof(keyValue));
-	config.SetString("Game", "Name", keyValue);
+	WritePlayerNameToIniFile();
+	WriteServerAddressListToIniFile();
```

Comment:
As a slight addendum, and not something I want to holdup this PR, but I was pondering if this is the optimal breakdown in terms of levels of abstraction.

The `config.SetString` method seems to be at about a similar level of abstraction as the new `WritePlayerNameToIniFile` method. The code to fetch a string from the user interface appears to be lower level code though. I was thinking it might make for a better logical breakdown if the buffering and `GetDlgItemText` was the part that got wrapped into a method.

This is of course trying to apply principles from that book you recommended a long while back.

On the other hand, the `WritePlayerNameToIniFile` method has that looping construct, which makes it seem higher level that what `config.SetString` offers.

Anyway, just food for thought. No changes required.

Reply:
Actually, I thought the same thing but went with this PR solution because it was quicker to implement than trying to abstract the GetDlgItemText and SetString functions. I was also thinking we could create a class that handled the .ini file access similar to what op2ext does long term.